components/header-bar/src/command-palette/hooks/use-navigation.js

```diff
@@ -0,0 +1,219 @@
+import { useCallback, useRef } from 'react'
+import { useCommandPaletteContext } from '../context/command-palette-context.js'
+
+export const GRID_ITEMS_LENGTH = 8
+
+export const useNavigation = ({ setShow, itemsArray, show }) => {
+    const modalRef = useRef(null)
+    const {
+        activeSection,
+        currentView,
+        filter,
+        highlightedIndex,
+        setHighlightedIndex,
+        setFilter,
+        setCurrentView,
+        setActiveSection,
+    } = useCommandPaletteContext()
+
+    const goToDefaultView = useCallback(() => {
+        setFilter('')
+        setCurrentView('home')
+        setActiveSection('grid')
+        setHighlightedIndex(0)
+    }, [setActiveSection, setCurrentView, setFilter, setHighlightedIndex])
+
+    const handleListViewNavigation = useCallback(
+        (event) => {
+            const lastIndex = itemsArray.length - 1
+            switch (event.key) {
+                case 'ArrowDown':
+                    setHighlightedIndex(
```

Comment:
are we sure we don't need preventDefault for ArrowDown and Up? (maybe ArrowDown scrolls the page?)